contracts/rules/solcompiler.bzl

```diff
@@ -0,0 +1,114 @@
+
+_SOLIDITY_COMPILER_BINARY_BUILD = """
+package(default_visibility = ["//visibility:public"])
+
+exports_files(
+  ['solc'],
+  visibility = ['//visibility:public'],
+)
+
+exports_files(
+  ['sol2go.sh'],
+  visibility = ['//visibility:public'],
+)
+
+filegroup(
+  name = 'file',
+  srcs = ['solc', 'sol2go.sh'],
+  visibility = ['//visibility:public'],
+)"""
+
+_SOLIDITY_COMPILER_BINARY_WIN64_BUILD = """
+package(default_visibility = ["//visibility:public"])
+
+exports_files(
+  ['solc.exe', 'msvcp140.dll'],
+  visibility = ['//visibility:public'],
+)
+
+exports_files(
+  ['sol2go.sh'],
+  visibility = ['//visibility:public'],
+)
+
+filegroup(
+  name = 'file',
+  srcs = ['solc', 'sol2go.sh', 'msvcp140.dll'],
+  visibility = ['//visibility:public'],
+)"""
+
+_SOLIDITY_SOL2GO = """
+#!/usr/bin/env bash
+
+ABIGEN="$2"
+OUTFILE="$3"
+PKG="$4"
+SOLCOMPILER=$5
+GO_ETHEREUM_IMPORTMAP=$6
+GO_ETHEREUM_REPO_BASE=$7
+GO_ETHEREUM_LABEL_NAME=$8
+
+EXTERNAL_DIR={external_dir}
+GO_ETHEREUM="$EXTERNAL_DIR/$GO_ETHEREUM_LABEL_NAME/"
+OUTDIR=$(dirname $OUTFILE)
+
+#Create src/github.com/ethereum/ tree needed by goimports
+mkdir -p $OUTDIR/src/$GO_ETHEREUM_REPO_BASE
+ln -f -s $GO_ETHEREUM  $OUTDIR/src/$GO_ETHEREUM_IMPORTMAP
+
+#Generate abi file
+$SOLCOMPILER --overwrite --bin --abi -o $OUTDIR $1
+CONTRACTNAME=$($SOLCOMPILER  --abi  $1  | sed -n 's,^=.*:\(.*\) .*=$,\\1,p')
```

Comment:
on mac, this value is empty. Not sure why yet...

Reply:
@prestonvanloon Hi, I managed to install a macos VM (didn't know it aws possible), I will try to fix the problem.